### PR TITLE
added credit card validation for android

### PR DESCRIPTION
**Pull request by arnsa**
This is a fix for an issue #46 for Android platform.

### CODE REVIEW

**Comment by arnsa** [comment]
#46

File: android/src/main/java/com/pw/droplet/braintree/Braintree.java [hunk 1]
```diff
@@ -59,6 +67,41 @@ public void onPaymentMethodNonceCreated(PaymentMethodNonce paymentMethodNonce) {
...
       });
+      this.mBraintreeFragment.addListener(new BraintreeErrorListener() {
+        @Override
+        public void onError(Exception error) {
+          if (error instanceof ErrorWithResponse) {
+            ErrorWithResponse errorWithResponse = (ErrorWithResponse) error;
+            BraintreeError cardErrors = errorWithResponse.errorFor("creditCard");
+            if (cardErrors != null) {
+              Gson gson = new Gson();
+              final Map<String, String> errors = new HashMap<>();
```
Comment (lkorth) [inline android/src/main/java/com/pw/droplet/braintree/Braintree.java]: There's no need to pull Gson in for just this, you could create a `JSONObject` here and just use `JSONObject#toString` to convert it to a string for the `nonceErrorCallback`.

**Comment by peterpme** [comment]
:wave: Anything I can do to help move this? @arnsa thanks for your help!

**Comment by peterpme** [comment]
:wave:  @kraffslol anything I can do to help?

**Comment by kraffslol** [comment]
I'll be pulling this, although I think @lkorth has a good idea about not using Gson for this. I'd really appreciate if someone could make a pull request using JSONObject instead for future versions. Sorry about the late response.

**Pull request merged**